### ISSUE TITLE
Consider transforming into a lerna monorepo

### ISSUE THREAD

**Issue by jbalsas** [bryceosterhaus/liferay-npm-scripts#3]
As we're starting to put together the scripts and tools necessary to build our liferay-portal projects, it might be a good idea to turn this into a lerna monorepo, so we can house different packages.

Other than this one, candidates to live alongside here could be:
- [liferay-npm-bundler-preset-liferay-dev](https://github.com/julien/liferay-npm-bundler-preset-liferay-dev). Already used by this project.
- [liferay-jest-junit-reporter](https://github.com/jbalsas/liferay-jest-junit-reporter). Only useful for our jest-jenkins integration. Might be needed if we also add a `--test` script.

We can think of the better release strategy, but keeping all contained should help make more sense of this as a whole.

What do you think?

**Comment by julien** [comment]
@jbalsas sounds good to me.

**Comment by julien** [comment]
@bryceosterhaus let me know when you want me to transfer the [liferay-npm-bundler-preset-liferay-dev](https://github.com/julien/liferay-npm-bundler-preset-liferay-dev) repo

**Comment by jbalsas** [comment]
Shouldn't be necessary, @julien! Just running [`lerna import`](https://github.com/lerna/lerna/tree/master/commands/import#readme) with a local clone is enough to import a regular repo into a lerna monorepo keeping the history.

**Comment by julien** [comment]
@jbalsas Ah ok, I didn't know that.
Thanks for the heads up

**Comment by bryceosterhaus** [comment]
I think it might make more sense to get moved to https://github.com/liferay/liferay-npm-build-tools since those are all very similar types of tools.

**Comment by jbalsas** [comment]
I was initially of that opinion, but @izaera kinda convince me otherwise. I'll let him weigh in :)

**Comment by izaera** [comment]
OK. So, what I proposed was to have two different repos, one for public things (`liferay-npm-build-tools`) and another one for internal tools (like `liferay-npm-scripts`). That way we don't introduce noise in the public repo for customers or people interested to use it. Also, we decouple both release lifecycles which will be, with a lot of probability, quite different.

**Comment by bryceosterhaus** [comment]
Ah I see. One thing to note is that I don't necessarily think `liferay-npm-scripts` should be entirely internal. Since its fairly simple and straight forward, I don't see any issue with being a public facing tool. For the release cycles though, that makes sense and I am on board for that.

**Comment by jbalsas** [comment]
So, based on what we've been discussing, what I think we could do to keep this moving forward would be:
- Keep this public. There's certainly no need to make it private in any way. It simply turns out that the focus of this (for now) is mostly internal.
- Rename this repo to something better matches its immediate intent... `liferay-dxp-build-tools`??
- Turn it into a lerna repo
- Import https://github.com/julien/liferay-npm-bundler-preset-liferay-dev
- Import https://github.com/jbalsas/liferay-jest-junit-reporter

I think at this point is clear that at least, every time we update either the `bundler-preset `or the `jest-junit` reporter, we will need to release an updated version of the `npm-scripts` which is the one we'd like to propagate to our dxp build environment. For that workflow, having those 3 repos together will simplify maintenance for all of us :)

What do you think?

**Comment by bryceosterhaus** [comment]
Sounds good to me. Only concern I would raise is the `liferay-dxp-build-tools` name. I'm not totally opposed to it, but I think we need to make it more  frontend centric to distinguish itself from any of our backend build tools.

**Comment by jbalsas** [comment]
I completely agree about the name... it was just the first thing that came to my mind. We can leave the renaming until the very end. I'm sure we'll have a better understanding of it once we see the bigger picture.

**Comment by bryceosterhaus** [comment]
Oh I see. I haven't really seen many modules outside of the ones in liferay-portal which is why I was confused at first.

**Comment by bryceosterhaus** [comment]
I used https://github.com/unravelin/tomono to merge the repos into a lerna repo [here](https://github.com/bryceosterhaus/liferay-npm-tools). Not totally sure if I did this correctly or if its the right pattern you were thinking. Let me know if this looks alright, @jbalsas.

**Comment by jbalsas** [comment]
Hey @bryceosterhaus, looks good to me! We'll need to transfer the issues there and delete this one, then?
